core/control-plane/transfer-core/src/main/java/org/eclipse/dataspaceconnector/transfer/core/command/handlers/SingleTransferProcessCommandHandler.java

```diff
@@ -56,9 +56,10 @@ public void handle(SingleTransferProcessCommand command) {
      * If the {@link TransferProcess} was indeed modified, implementors should return {@code true}, otherwise {@code false}
      *
      * @param process The {@link TransferProcess}
+     * @param command The {@link SingleTransferProcessCommand}
      * @return true if the process was actually modified, false otherwise.
      */
-    protected abstract boolean modify(TransferProcess process);
+    protected abstract boolean modify(TransferProcess process, T command);
```

Comment:
This is not needed - only the `FailTransferCommandHandler` requires access to the command object, and that can be achieved by hooking into the `handle` method:

```java
// in FailTransferCommandHandler
@Override
public void handle(FailTransferCommand command) {
        // store for later use
        this.failCommand = command;
        // delegate back up the chain
        super.handle(command);
    }

@Override
protected boolean modify(TransferProcess process){
        // use this.failCommand
}
```

Reply:
Isn't the handler registered only once at startup time?

Reply:
yes, the handler is registered only once, but the `handle` method should be invoked whenever the handler receives a command

Reply:
Yes but saving state between handle calls could lead to strange bug due being the same instance right?

Reply:
the state, i.e. the command, would only be valid between `handle` and `modify`, and should be set to `null` after that, that's correct.

Reply:
@paullatzelsperger I don't think it's a good idea to make the `CommandHandler` stateful, I like the @wolf4ood solution more, we expect that the command brings some information needed to be stored in the entity, so having the command passed to the `modify` method makes total sense.

Reply:
@ndr-brt I don't quite understand what the problem is in carrying the command over from `handle()` to `modify()` granted that it gets nullified at the end of `modify()`. It seems much less intrusive than adding a method parameter that is only used in one particular situation.